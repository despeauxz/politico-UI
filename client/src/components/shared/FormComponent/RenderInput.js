import React, { Fragment, Component } from 'react';
import PropTypes from 'prop-types';


class RenderInput extends Component {
  static propTypes = {
      id: PropTypes.string.isRequired,
      value: PropTypes.oneOfType([PropTypes.string, PropTypes.bool, PropTypes.number]),
      label: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      type: PropTypes.string.isRequired,
      className: PropTypes.string.isRequired,
      error: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
      required: PropTypes.bool,
      placeholder: PropTypes.oneOfType([PropTypes.string, PropTypes.bool, PropTypes.number]),
      handleChange: PropTypes.func.isRequired,
      onFocus: PropTypes.func
  };

  static defaultProps = {
      value: '',
      errors: {}
  };

  render() {
      const {
          id, type, name, placeholder, className,
          value, error, required, onFocus, ...rest
      } = this.props;
      
      return (
          <Fragment>
              <div className="form-group">
                  <label htmlFor={id} className="control-label">
                      {required && <span className="danger">* </span>}
                      {this.props.label}
                  </label>

                  <input
                      id={id}
                      type={type}
                      name={name}
                      placeholder={placeholder}
                      className={className}
                      value={value}
                      onChange={e => this.props.handleChange(e)}
                      onFocus={onFocus}
                      error={error}
                      {...rest}
                  />
                  {error && <div className="text-red-600 text-xs text-left">{error}</div>}
              </div>
          </Fragment>
      );
  }
}


export default RenderInput;
