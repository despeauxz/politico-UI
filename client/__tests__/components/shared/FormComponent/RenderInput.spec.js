import React from 'react';
import RenderInput from '@components/shared/FormComponent/RenderInput';

const props = {
    name: 'name',
    label: 'Name',
    placeholder: 'Name',
    type: 'text',
    error: "",
    value: '',
    className: '',
    id: 'name',
    handleChange: jest.fn(),
};

describe('Render Input', () => {
    it('should render the RenderInput component when used', () => {
        const shallowWrapper = shallow(<RenderInput {...props} />);

        expect(toJson(shallowWrapper)).toMatchSnapshot();
        expect(shallowWrapper.find('RenderInput')).toBeTruthy();
    });

    it('should call onChange prop', () => {
        const onChange = jest.fn();

        const component = mount(<RenderInput {...props} handleChange={onChange} />);
        component.find('input').simulate('change');
        expect(onChange).toHaveBeenCalled();

        component.unmount();
    });
});
