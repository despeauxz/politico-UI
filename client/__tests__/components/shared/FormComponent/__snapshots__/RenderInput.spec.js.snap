// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Input should render the RenderInput component when used 1`] = `
<Fragment>
  <div
    className="form-group"
  >
    <label
      className="control-label"
      htmlFor="name"
    >
      Name
    </label>
    <input
      className=""
      id="name"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
  </div>
</Fragment>
`;
